sklearn/utils/estimator_checks.py

```diff
@@ -131,6 +131,29 @@ def _yield_classifier_checks(name, Classifier):
     if 'class_weight' in Classifier().get_params().keys():
         yield check_class_weight_classifiers
 
+def check_supervised_y_no_nan(name, Estimator):
+    # Checks that the Estimator targets are not NaN.
+
+    warnings.simplefilter("ignore")
+    X = np.random.randn(10, 5)
+    y1 = np.random.randn(10) / 0.
```

Comment:
It will not matter in the absolute sense. But it is better to have deterministic input for every run (both X and y1)